core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1367,10 +1373,6 @@ WorkflowModel decide(WorkflowModel workflow) {
                 return decide(workflow);
             }
 
-            if (!outcome.tasksToBeUpdated.isEmpty() || !tasksToBeScheduled.isEmpty()) {
-                executionDAOFacade.updateWorkflow(workflow);
-            }
-
             return workflow;
```

Comment:
Couple of issues:
1. the fix is not going to work for cases the set_variable task is not the first task of the workflow
2. hashcode computation of workflow on every execution is not going to be efficient
3. remove block (i.e. line 1370-1373 above) from the decide method could cause side effect

Reply:
Hi @jxu-nflx , Updated as per the comment.